packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -200,22 +200,26 @@ distributionUrl=https\\://services.gradle.org/distributions/gradle-$gradleVersio
   }
 }
 
-/// Overwrite android/local.properties in the specified Flutter project, if needed.
+/// Overwrite local.properties in the specified Flutter project's Android
+/// sub-project, if needed.
 ///
-/// Throws, if `pubspec.yaml` or Android SDK cannot be located.
+/// Throws tool exit, if `pubspec.yaml` is invalid.
 ///
-/// If [requireSdk] is `true` this will fail with a tool-exit if no Android Sdk
+/// If [requireSdk] is `true` this will fail with a tool exit if no Android Sdk
 /// is found.
 Future<void> updateLocalProperties({
   @required FlutterProject project,
   BuildInfo buildInfo,
   bool requireAndroidSdk = true,
 }) async {
+  if (project.manifest == null) {
+    throwToolExit('Invalid `pubspec.yaml`');
```

Comment:
Perhaps we should never be able to create a FlutterProject with a null manifest?

Reply:
Right. I'll push that to a separate PR.